libr/bin/p/bin_elf.inc.c

```diff
@@ -195,15 +198,14 @@ static void process_constructors(RBinFile *bf, RList *ret, int bits) {
 		if (type == -1) {
 			continue;
 		}
-		if (sec->size > 0xffffff) {
-			continue;
-		}
 		ut8 *buf = calloc (sec->size, 1);
 		if (!buf) {
 			continue;
 		}
+
 		(void)r_buf_read_at (bf->buf, sec->paddr, buf, sec->size);
```

Comment:
check return to realloc the buffer and maybe add a warning if that section->size needs to be trimmed to fit the reality

Reply:
can you be more clear what needs to happen here?
```
int size = r_buf_read_at(...)
if (size != sec->size) {
        if (size < sec->size) { R_LOG_WARN ("unexpected section size"); }
        realloc (buf, size);
        sec->size = size;
}
```
something like this?

Reply:
Yep like this